### PR TITLE
Add to List Users API the ability to filter users who need or want to provide mentoring (issue # 424)

### LINKED ISSUE #424: Add to List Users API the ability to filter users who need or want to provide mentoring
## Description
As a member of the app,
I need be able to filter other members for their availability,
so that I can choose the member I want to create a mentorship relation with.

APIs used to fetch Users, don't have an option to filter users by `need_mentoring` and `available_to_mentor` attributes.

### Update [Required]
- [ ] Add a query parameter support to `GET /users` and `GET /users/verified`, to filter users by `need_mentoring` and `available_to_mentor` attributes.
- [ ] Update Swagger UI documentation for this API.
- [ ] Write at least one test to prove your new feature is working.

## Definition of Done
- [ ] All of the required items are completed.
- [x] Approval by 1 mentor.

## Estimation
4 hours

### PR DESCRIPTION
**Pull request by lmcicat**
### Description
As a member of the app,
I need be able to filter other members for their availability,
so that I can choose the member I want to create a mentorship relation with.

APIs used to fetch Users, don't have an option to filter users by need_mentoring and available_to_mentor attributes.

Fixes #424

### Type of Change:

- Code
- Quality Assurance
- Documentation

**Code/Quality Assurance Only**
- This change requires a documentation update (software upgrade on readme file)
- New feature (non-breaking change which adds functionality pre-approved by mentors)

### How Has This Been Tested?
We created more unit tests in the tests/users/test_api_list_users.py. You can run these tests by following this guideline: https://github.com/anitab-org/mentorship-backend#run-tests.

### Checklist:

- [x] My PR follows the style guidelines of this project
- [x] I have performed a self-review of my own code or materials
- [x] I have commented my code or provided relevant documentation, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] Update Swagger documentation and the exported file at /docs folder

**Code/Quality Assurance Only**
- [x] My changes generate no new warnings \
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Review by faznan3nazer — COMMENTED** [review]
Tested the feature locally.

**Comment by lmcicat** [comment]
Just realized I never updated the `list_users` docs to include the new parameters. pushed up the change now.

**Review by foongminwong — APPROVED** [review]
Great job @lmcicat and @brianguo98!!! I apologize for the late review, just want to make sure the PR is tested correctly before merging..
@isabelcosta here're the GIFs of before (original) vs after pr535

**Before pr535 - `Original`**
before_pr535

**After pr535 - `GET /users`**
after_pr535_GETusers

**After pr535 - `GET /users/verified`**
after_pr535_GETusersverified

**Review by foongminwong — COMMENTED** [review]
@isabelcosta I wasn't able to run the unit test `test_api_list_users.py` since I'm still having issue that I'm tryign to fix [here](https://anitab-org.zulipchat.com/#narrow/stream/216325-quality-assurance/topic/Testing.20Error/near/195101602)

**Comment by isabelcosta** [comment]
@faznan3nazer @foongminwong thank you sooooo much for your QA work here! it is super duper helpful :) 🎉

**Review by isabelcosta — COMMENTED** [review]
I ran the code locally, and I tested the following use cases:
1. invalid `available_to_mentor` value set to "urusla" which is not a boolean

image

**Comment by lmcicat** [comment]
@isabelcosta what do you want to happen if true/false aren't passed in?

**Review by SanketDG — CHANGES_REQUESTED** [review]
I think keeping a flag value as string is going to cause a lot of problems with random data and tests, so a better way of enforcing this would be to use boolean as the data type strictly in all places with the `false` value as default!

Now for when we will consume this API on the client side, we can just set default values there!

**Comment by isabelcosta** [comment]
I'm sorry for the delay in my response @lmcicat .
Would you be willing to address @SanketDG 's comment to improve this PR and fix the merge conflicts?
If not, that's ok we can leave this to another contributor, if yes, we can wait.

**Comment by lmcicat** [comment]
Sorry for the delayed response @isabelcosta @SanketDG . I am pretty busy with work at the moment, but I can work on it come Friday / the weekend. If that isn't soon enough for you, feel free to let other contributors work on it :)

**Comment by isabelcosta** [comment]
no problem @lmcicat ;) It's ok we can wait

**Comment by lmcicat** [comment]
hi @isabelcosta I'm sorry about this, but I am much more busy with work than I expected. If you could open this up to others to grab, that would probably be a good idea.

**Comment by isabelcosta** [comment]
@lmcicat no worries at all, I will open to other contributors, if no one picks up and you want to contribute in the future, feel free to do so! Thank you for your patience and work I really really appreciate it. 🙌  This can help a future contributor 🤗 

Good luck with your work!

**Pull request closed without merging**
